### PR TITLE
[Core] Missing copy data clone on element and condition base

### PR DESCRIPTION
**Pull request by loumalouomega**
Besides, we should add the data copy to the derived ones too

### CODE REVIEW

File: kratos/includes/condition.h [hunk 1]
```diff
@@ -264,8 +264,11 @@ class Condition : public GeometricalObject, public Flags
     virtual Pointer Clone (IndexType NewId, NodesArrayType const& ThisNodes) const
     {
         KRATOS_TRY
-	std::cout<<" Call base class condition Clone "<<std::endl;
-        return Condition::Pointer(new Condition(NewId, GetGeometry().Create(ThisNodes), pGetProperties()));
+	KRATOS_INFO("Condition") << " Call base class condition Clone " << std::endl;
+        Condition::Pointer p_new_cond = Kratos::make_shared<Condition>(NewId, GetGeometry().Create(ThisNodes), pGetProperties());
+	p_new_cond->Data() = this->Data();
+	p_new_cond->Set(Flags(*this));
```
Comment (pooyan-dadvand) [inline kratos/includes/condition.h]: I would suggest `static_cast` here...

File: kratos/includes/element.h [hunk 2]
```diff
@@ -262,8 +262,11 @@ class Element : public GeometricalObject, public Flags
     virtual Pointer Clone (IndexType NewId, NodesArrayType const& ThisNodes) const
     {
         KRATOS_TRY
-	std::cout<<" Call base class element Clone "<<std::endl;
-        return Element::Pointer(new Element(NewId, GetGeometry().Create(ThisNodes), pGetProperties()));
+	KRATOS_INFO("Element") << " Call base class element Clone " << std::endl; 
+        Element::Pointer p_new_elem = Kratos::make_shared<Element>(NewId, GetGeometry().Create(ThisNodes), pGetProperties()); 
+        p_new_elem->Data() = this->Data(); 
+        p_new_elem->Set(Flags(*this));
```
Comment (pooyan-dadvand) [inline kratos/includes/element.h]: And here...

**Review by pooyan-dadvand — CHANGES_REQUESTED** [review]
Your proposal make sense as clone should really clone everything. Nevertheless this clone method was an special one for prototyping where we haven't used the other data of the elements. 

Why you have not change this ones in the Clone of GeometricalObject?

**Comment by loumalouomega** [comment]
There is not clone in GeometricalObject

File: kratos/includes/condition.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/includes/condition.h]: What do you mean with static_cast

**Comment by pooyan-dadvand** [comment]
So I was saying to add one and put the repeated core there.

**Comment by loumalouomega** [comment]
But the GeometricalObject just contains the geometry. Does not contain Data or Flags

**Comment by loumalouomega** [comment]
OK, I got it, to add the clone to geometrical object, so in the clone does not generate a base geometry, that explains the error I was getting and why was unable to plot the mesh

**Comment by loumalouomega** [comment]
I cannot add the Clone to the GeometricalObject, it conflicts with the Clone of the Condition and the Element

**Comment by pooyan-dadvand** [comment]
why it makes conflict?

**Comment by loumalouomega** [comment]
Because the geometrical object lacks of flags and data container. So in
case you define the clone it tries to override, but it lacks that, so can
not compile. Maybe adding a clone method to the geometrical object, but not
virtual will work,but I don't know if this was the idea.

**Comment by pooyan-dadvand** [comment]
Would you please add a test to it? A cpp test would be more appropated

**Comment by loumalouomega** [comment]
Ok

**Comment by loumalouomega** [comment]
Done @pooyan-dadvand

**Review by pooyan-dadvand — APPROVED** [review]
👍

**Pull request merged**